src/Compilers/VisualBasic/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.vb

```diff
@@ -287,6 +298,75 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             End Select
         End Sub
 
+        ''' <summary>
+        ''' Returns true if tuple type syntax can be used to refer to the tuple type without loss of information.
+        ''' For example, it cannot be used when extension tuple is using non-default friendly names. 
+        ''' </summary>
+        ''' <param name="tupleSymbol"></param>
+        ''' <returns></returns>
+        Private Function CanUseTupleTypeName(tupleSymbol As INamedTypeSymbol) As Boolean
+            Dim currentUnderlying As INamedTypeSymbol = tupleSymbol.TupleUnderlyingType
+
+            While currentUnderlying.Arity = TupleTypeSymbol.RestPosition
+                tupleSymbol = DirectCast(currentUnderlying.TypeArguments(TupleTypeSymbol.RestPosition - 1), INamedTypeSymbol)
+                Debug.Assert(tupleSymbol.IsTupleType)
+
+                If HasNonDefaultTupleElementNames(tupleSymbol) Then
+                    Return False
+                End If
+
+                currentUnderlying = tupleSymbol.TupleUnderlyingType
+            End While
+
+            Return True
+        End Function
+
+        Private Shared Function HasNonDefaultTupleElementNames(tupleSymbol As INamedTypeSymbol) As Boolean
+            Dim elementNames = tupleSymbol.TupleElementNames
+
+            If Not elementNames.IsDefault Then
+                For i As Integer = 0 To elementNames.Length - 1
+                    If (elementNames(i) <> TupleTypeSymbol.TupleMemberName(i + 1)) Then
+                        Return True
+                    End If
+                Next
+            End If
+
+            Return False
+        End Function
+
+        Private Sub AddTupleTypeName(symbol As INamedTypeSymbol)
+            Debug.Assert(symbol.IsTupleType)
+
+            Dim elementTypes As ImmutableArray(Of ITypeSymbol) = symbol.TupleElementTypes
+            Dim elementNames As ImmutableArray(Of String) = symbol.TupleElementNames
+            Dim hasNames As Boolean = Not elementNames.IsDefault
+
+            AddPunctuation(SyntaxKind.OpenParenToken)
+
+            Dim first As Boolean = True
+            For i As Integer = 0 To elementTypes.Length - 1
+                If Not first Then
```

Comment:
changed both VB and C# implementations

---
In reply to: [71446570](https://github.com/dotnet/roslyn/pull/12596#discussion_r71446570) [](ancestors = 71446570)